docs/content/reference/resolvers.md

```diff
@@ -0,0 +1,203 @@
+
+This is also very common use case that comes up where we want to bind a graphQL field to a Go struct method
+
+```go
+type Person {
+    Name string
+}
+
+type Car struct {
+    Make string
+    Model string
+    Color string
+    OwnerID *string
+    OdometerReading int
+}
+
+func (c *Car) Owner() (*Person) {
+    // get the car owner
+    //....
+    return owner
+}
+```
+
+And then in your graphQL schema:
+```graphql
+type Car {
+    make: String!
+    model: String!
+    color: String!
+    odometerReading: Int!
+    owner: Person
+}
+```
+
+And in the gqlgen config file:
+```yaml
+models:
+    Car:
+        model: github.com/my/app/models.Car
+    Person:
+        model: github.com/my/app/models.Person
+```
+
+Here, we see that there is a method on car with the name ```Owner```, thus the ```Owner``` function will be called if
+a graphQL request includes that field to be resolved
+
+## Bind when the field names do not match
+
+There are two ways you can bind to fields when the the Go struct and the graphQL type do not match.
+
+
+The first way is you can bind resolvers to a struct based off of struct tags like the following:
+
+```go
+type Car struct {
+    Make string
+    ShortState string 
+    LongState string `gqlgen:"state"`
+    Model string
+    Color string
+    OdometerReading int
+}
+```
+
+And then in your graphQL schema:
+```graphql
+type Car {
+    make: String!
+    model: String!
+    state: String!
+    color: String!
+    odometerReading: Int!
+}
+```
+
+And in the gqlgen config file add the line:
+```yaml
+struct_tag: gqlgen
+
+models:
+    Car:
+        model: github.com/my/app/models.Car
+```
+
+Here even though the graphQL type and Go struct have different field names, there is a Go struct tag field on ```longState```
+That matches and thus ```state``` will be bound to ```LongState```
+
+
+The second way you can bind fields is by adding a line into the config file such as:
+```go
+type Car struct {
+    Make string
+    ShortState string 
+    LongState string
+    Model string
+    Color string
+    OdometerReading int
+}
+```
+
+And then in your graphQL schema:
+```graphql
+type Car {
+    make: String!
+    model: String!
+    state: String!
+    color: String!
+    odometerReading: Int!
+}
+```
+
+And in the gqlgen config file add the line:
+```yaml
+models:
+    Car:
+        model: github.com/my/app/models.Car
+        fields:
+            state:
+                fieldName: LongState
+```
+
+## Binding to Anonymous or Embedded Structs
+All of the rules from above apply to a struct that has an embedded struct.
+
+```go
+type Truck {
+    car
+
+    is4x4 bool
+}
+
+type Car struct {
+    Make string
+    ShortState string 
+    LongState string
+    Model string
+    Color string
+    OdometerReading int
+}
+```
+
+And then in your graphQL schema:
+```graphql
+type Truck {
+    make: String!
+    model: String!
+    state: String!
+    color: String!
+    odometerReading: Int!
+    is4x4: Bool!
+}
+```
```

Comment:
May be worth mentioning embedding is also a good way to create thin presenters over database types without needing resolvers, eg

```
type Cat struct {
    db.Cat
}

func (c *Cat) ID() string {
     return fmt.Sprintf("%d:%d", c.Shard, c.Id)
}
```